### DISCUSSION TITLE
AD Editor Update to v1.1.1.4

### DISCUSSION THREAD

**Discussion by KillBait** [Stephan-S/FS19_AutoDrive#2047] [General]
I've been working on updating the AD Editor to support AD 1.1.1.4

While this was for my own personnel use, i was wondering if anybody else has any interest in using it.

I've changed and updated a few things,  so i'll quickly list them here.

General Adjustments
- Adjusted the path colours to match the ingame editor

Fixes/Additions for AD 1.1.1.4
- saving a config doesn't mess up the primary/secondary paths
- Correctly displays the lower priority one way + dual way routes
- Added a button to swap a node between primary/secondary route
- You can directly added a secondary route node.
-
An example of the changes - Let me know what you think

![AD Editor Alpha update](https://user-images.githubusercontent.com/7914589/128632370-731faffb-3964-465a-b950-a35b51d623aa.png)

The only thing i have left todo on my list is add the colours for reverse routes, but my java
knowledge is very basic, so it may take me a while to figure it out.

**Comment by Axel32019** [comment]
Hello @KillBait, 
Thanks for your contribution! 
I am also not so fluent in java, but from a rought look I would merge it to master branch. 
If you, @Iwan1803 or @Williwillswisse  find users for testing please point them to the beta branch.

The reverse connections are stored in a special way, so may be this snippet will help to understand:
```
function ADGraphManager:isReverseRoad(start, target)
    ....
    return (not table.contains(target.incoming, start.id) and table.contains(start.out, target.id))
end

```

**Reply by KillBait** [reply]
Hi @Axel32019 

Thanks for the code hint, was able to get them showing properly now.

I'm working on a few other things but the main one is GUI related since the initial window size was getting a bit too large
due to all the buttons. I have swapped the buttons for icons and separated the different types out, plus i want to add a 
menu so its possible to save a config with a different name

Currently it looks like this (excuse the horrible icons :-( i'm terrible at anything art related)

If anybody is willing to create a new set of icons and donate them to the project, i would be most grateful.

![Shot_New](https://user-images.githubusercontent.com/7914589/130446700-55479dd6-f705-422f-b169-57b55a281159.png)

**Reply by Axel32019** [reply]
@KillBait Could you please be so kind and provide your update as PR, including the compiled .jar file?

**Reply by KillBait** [reply]
No problem.. Creating reverse connections/menu/marker groups are not finished yet but I can remove them from the PR and do another when they are complete

**Reply by KillBait** [reply]
I have creating reverse connections working and the menu's.. i wanted to get those done before i did the Pull Request.. it'll be a day or two before i get around to the PR due to work commitments
